app.py

```diff
@@ -36,45 +36,60 @@
+    """Load or initialize admin settings from a JSON file.
+
+    If no setting file exists, it writes default loaded_settings and returns them.
+
+    If a setting file exists, it reads and returns the updated loaded_settings.
+
+    :returns: admin settings from file or default settings
+    """
+
+    # configure default admin settings
+    config_folder = app.config['CONFIG_FOLDER_PATH']
+    settings_file_path = path.join(config_folder, 'admin_settings.json')
+    default_settings = {
         'banner': {
-            'banner_enabled': False,
-            'banner_importance': False,
-            'banner_message': ''
+            'enabled': False,
+            'importance': False,
+            'message': ''
         },
         'YODA_THEME': app.config.get('YODA_THEME')
     }
 
     try:
         # If file doesn't exist, create and write the default configuration
-        if not path.exists(config_file_path):
-            with open(config_file_path, 'w') as file:
-                json.dump(default_config, file)
-            return default_config
-
-        # If the file exists, read and return the configuration
-        with open(config_file_path, 'r') as file:
-            settings = json.load(file)
-            banner_set = settings.get('banner', default_config['banner'])  # Get banner settings or use default
-            return {
+        if not path.exists(settings_file_path):
+            makedirs(config_folder, exist_ok=True)
+            with open(settings_file_path, 'w') as file:
+                json.dump(default_settings, file)
+            return default_settings
+
+        # If the file exists, read and return the setting
+        with open(settings_file_path, 'r') as file:
+            loaded_settings = json.load(file)
+            merged_settings = {
                 'banner': {
-                    'banner_enabled': banner_set.get('banner_enabled', default_config['banner']['banner_enabled']),
-                    'banner_importance': banner_set.get('banner_importance', default_config['banner']['banner_importance']),
-                    'banner_message': banner_set.get('banner_message', default_config['banner']['banner_message'])
+                    **default_settings['banner'],
+                    **loaded_settings.get('banner', {})
                 },
-                'YODA_THEME': settings.get('YODA_THEME', default_config['YODA_THEME'])
+                'YODA_THEME': loaded_settings.get('YODA_THEME', default_settings['YODA_THEME'])
             }
+            return merged_settings
+    except PermissionError:
+        print("Permission denied while accessing settings file.")
+    except FileNotFoundError:
+        print("Settings file not found.")
+    except json.JSONDecodeError:
+        print("Error decoding JSON from the settings file.")
     except Exception:
-        print("An unexpected error occurred")
-        return default_config
-
+        print("Unexpected error occurred.")
```

Comment:
I am thinking if I shall keep the print statements, since I am not sure they'll be helpful for users. Perhaps only for the developers to diagnose possible errors?

Reply:
Keep them for troubleshooting, but use log_error from util.

Reply:
Improved